src/Screens/Main/SearchMentor.tsx

```diff
@@ -0,0 +1,330 @@
+import React from 'react';
+import RN from 'react-native';
+import { useSelector } from 'react-redux';
+import * as RD from '@devexperts/remote-data-ts';
+
+import * as navigationProps from '../../lib/navigation-props';
+
+import fonts from '../components/fonts';
+import Message from '../components/Message';
+import * as localization from '../../localization';
+
+import MessageButton from '../components/MessageButton';
+import TextButton from '../components/TextButton';
+import TitledContainer from '../components/TitledContainer';
+import { textShadow } from '../components/shadow';
+import colors, { gradients } from '../components/colors';
+
+import * as mentorState from '../../state/reducers/mentors';
+import RemoteData from '../components/RemoteData';
+import { SearchMentorResultsRoute } from './SearchMentorResults';
+
+import CreatedBySosBanner from '../components/CreatedBySosBanner';
+
+export type SearchMentorRoute = {
+  'Main/SearchMentor': {};
+};
+
+type Props = navigationProps.NavigationProps<
+  SearchMentorRoute,
+  SearchMentorResultsRoute
+>;
+
+export default ({ navigation }: Props) => {
+  const [selectedSkills, selectSkill] = React.useState<string[]>([]);
+  const [skillSearch, setSkillSearch] = React.useState('');
+
+  const fetchMentors = () => {};
+
+  const skillsList = RD.remoteData.map(
+    useSelector(mentorState.get),
+    mentors =>
+      mentors
+        .map(mentor => mentor.skills) // collect all skills
+        .flat() // flatten results into one array
+        .filter((item, index, self) => self.indexOf(item) === index) // remove duplicates
+        .filter(e =>
+          skillSearch
+            ? e.toLowerCase().includes(skillSearch.toLowerCase())
+            : true,
+        ) // skill searching
+        .sort(), // simple alphabetical sort
+  );
+
+  const onPressSearch = () => {
+    navigation.navigate('Main/SearchMentorResults', {
+      skills: selectedSkills,
+    });
+  };
+
+  const onPressBack = () => {
+    navigation.goBack();
+  };
+
+  const onPressSkill = (item: any) => {
+    let s = selectedSkills.slice();
+    if (s.includes(item)) {
+      const filteredAry = s.filter(e => e !== item);
+      selectSkill(filteredAry);
+    } else {
+      s.push(item);
+      selectSkill(s);
+    }
+  };
```

Comment:
Array ei oo paras tietorakenne, jos halutaan toglailla juttuja. Setti vois natsaa paremmin.

Nää on liian kompleksit lainit

Reply:
For simplicity we decided to go with array.